### DISCUSSION TITLE
CVE-2022-42920 in apache bcel

### DISCUSSION THREAD

**Discussion by karel1980** [spotbugs/spotbugs#2251] [General]
Hi. Just wanted to let you know about this CVE: https://nvd.nist.gov/vuln/detail/CVE-2022-42920.
The good news is renovate[bot] already bumped bcel to 6.6.1 on nov 3rd
Given the severity of this CVE you might want to release 2.7.4.

Cheers!

**Comment by iloveeclipse** [comment]
Should be irrelevant for spotbugs, we don't **write** anything using BCEL, we only **read**

To be clear, see
* Original BCEL bug https://issues.apache.org/jira/browse/BCEL-363
* BCEL fix: https://github.com/apache/commons-bcel/pull/147

**Comment by karel1980** [comment]
I understand. I got here because this trips up build pipelines for projects using the owasp dependency checker plugin :) It's not a big deal, pinning bcel to 6.6.1 works fine as a workaround

**Reply by kelvinn** [reply]
Exact same boat here - looking forward to 2.7.4 :)

**Reply by karel1980** [reply]
@kelvinn We've verified that both adding the suppression for the owasp dependency checker and overriding the bcel version via dependencyManagement (in gradle) works.
HTH

**Comment by kiranprakash154** [comment]
Looks like the bcel was updated [here](https://github.com/spotbugs/spotbugs/commit/59eeb2d7e999b70f9bb1ff4639855a7742750eec)

Looking forward to the release of 2.7.4, When will that be?

Thanks in advance!

**Reply by karel1980** [reply]
4.7.0 through 4.7.3 were released between May and October. So at the current rate it could be expected somewhere in December :) - no guarantees though.

**Reply by PrasannaKumarGowdru** [reply]
Hi @karel1980  Even we are getting same security vulnerability CVE-2022-42920 from latest spot bug version, please let us know when the new version of Spot Bug will be available to fix this issue ( Our Management  team wants to know tentative date for this fix)

Thanks a lot in advance  :)

**Reply by karel1980** [reply]
I hope this question was not directed at me :) I am not affiliated with the project

**Comment by liwamdaman** [comment]
A little late to this thread, but also was curious about the release of 4.7.4, related to this bcel version bump. Hoping it comes soon!

Perhaps @iloveeclipse you could let us know of a tentative release date? Thanks 😄

**Comment by binkley** [comment]
When will this be resolved?
I have tried force updating the bcel dependency to latest (6.7.0), however that produces complaints:

```
The following errors occurred during analysis:
  Exception analyzing demo.TheFooTest using detector edu.umd.cs.findbugs.detect.FindUseOfNonSerializableValue
    org.apache.bcel.classfile.ClassFormatException: Invalid method signature: java/lang/NullPointerException
      At org.apache.bcel.classfile.Utility.typeOfSignature(Utility.java:1284)
```

**Reply by JuditKnoll** [reply]
The currently latest, 6.7.0 bcel has a bug, which breaks the spotbugs build. There is already a fix PR for it (https://github.com/apache/commons-bcel/pull/221), so hopefully it will be in the next bcel release. Until then you can use bcel 6.6.1 with spotbugs.

**Comment by smilysharma-kr** [comment]
how to use bcel 6.x along with spot bugs in gradle.. if I explicitly mention bcel, it is still referring to bcel 5.X

**Reply by hazendaz** [reply]
Please make sure if not already ticket is open on the gradle plugin to just override it to 6.6.1.  If you can raise PR there, I can make a release happen to solve it.

**Comment by hazendaz** [comment]
Going to close as we fixed in this repo already, just not released and it can be overwritten by downstreams.  For maven, it was done and released [here](https://github.com/spotbugs/spotbugs-maven-plugin/releases/tag/spotbugs-maven-plugin-4.7.3.5).  For gradle, I would suggest raising ticket over there to override it.  I looked but don't see it but think @KengoTODA could quickly get a release similar to what I did with maven.  That would then satisfy the issue but otherwise issues is addressed here so no point keeping ticket open.

**Reply by binkley** [reply]
@hazendaz Very happy to see this resolved -- and on my birthday, no less!
I overlong hesitated to recheck, and now looking again, the CVE error is gone from my builds.
Thank you.
